### ISSUE TITLE
[Issue] Cannot read property 'token' of undefined

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#430]
In GitLab by @PalaNolho on Jul 11, 2021, 19:37

I'm getting this error when triggering an ability that has DAE on it (ex: Rage)
This happens when I "roll" the ability and happens regardless if I have the token selected or not.
I was expecting that when I use the ability the DAE would "activate" but it stays in "inactive".

I believe it may be somehow related to `BetterRolls5e`and I noticed it this week.
Yesterday, when I clicked the ability, I would see an "Apply effect" button on the chat but today, after updating BetterRolls, I no longer see that button but the error persists.

![image](https://my-gitlab-bucket.s3.amazonaws.com/640711674/8d37319a5f3df6a995a74e9b86cca14d01259dce804887fbe9f5b685b9f5d326/image.png)

**Comment by tposney** [comment]
Can you confirm the version of better rolls and midi you are using?

**Comment by tposney** [comment]
I could only reproduce something like this if there is no token for the actor on the map. If there is a token I can't repro, so I'll need more information. At the least the version numbers I mentioned above, the actor that is having the problem plus you midi-qol settings.

**Comment by tposney** [comment]
In GitLab by @PalaNolho on Jul 12, 2021, 24:00

Here is the info you asked for: 
```
Mid-qol v0.8.31
BetterRolls5e v1.5.5
```
I'm also attaching the actor I was testing. 
Try the `Rage` feature.

(**Note:** Please be aware that the rage ability has a DAE associated, which is the one I was trying to activate by rolling the feature)

[fvtt-Actor-ulfgar-stormbearer.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/6e0be7934c94fd62c289cdd6cc5987acfa50adfe421e0ed692288237c41158e0/fvtt-Actor-ulfgar-stormbearer.json)

**Comment by tposney** [comment]
Ok, I loaded the actor and looked at the rage feature. The feature is not setup correctly - it is marked as a transfer effect, which means it is applied when the item is equipped, rather than when it is rolled. I attach  a version that works for me.

As I mentioned there was a midi problem when trying to activate effects when no token existed for the actor - I've fixed that - so it's possible the issue you were seeing was related to that, but you did mention that the problem occurred if the token was selected or not, so that might not be the issue.

I could not repro the problem except for the no token case.

If using the supplied item does not fix the problem, I'll need you to export you midi-settings which also shows me which modules are installed.

[fvtt-Item-rage__1_.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/c15d6dddcdde00a76ea04af82f082131ecb55411a3757e677f5a060ac69dfa4b/fvtt-Item-rage__1_.json)

**Comment by tposney** [comment]
In GitLab by @PalaNolho on Jul 12, 2021, 18:51

With your feature it seems to work. It shows the "apply" button and when I click it activates on the actor.

Out of curiosity, why did you use the `flags dae.DamageBonusMacro(*)` effect instead of the `Bonus Melee Weapon Damage (*)`

**Comment by tposney** [comment]
Just fyi, the only thing I changed in the item was to make it a non-transfer effect (so it would get applied when you roll the item), other than that there was no change. So really, pretty minor.

I just copied the rage item you had on your character, so did not give it any thought. Looking at the macro I think it is to make the damage scale with the barbarian levels in an easier way.

Damage bonus macro vs data.bonuses.damage.mwak/rwak is a 50/50 choice. If there are level based calculations it's probably easier to do in the damage bonus macro, rather than a fixed bonus, and damage bonus macros are more flexible. There is not much to choose between the two however.
